### PR TITLE
Adding rotateable wedge

### PR DESCRIPTION
**Pull request by Shimwell**
## Proposed changes

This adds a new parametric shape, a truncated triangle useful for cutting port and other gaps. The shape requires 3 lengths and a pivot point and optionally a pivot angle

Screenshot from 2020-12-11 23-34-21
truncated triangle(1)

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Code refactoring
- [ ] Documentation Update (if none of the other choices apply)
- [x] New tests

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] Pep8 applied
- [ ] Unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: paramak/parametric_components/truncated_triangle.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+            the shape by on the XY plane. Defaults to 0.
+        stp_filename (str, optional): defaults to "TruncatedTriangle.stp".
+        stl_filename (str, optional): defaults to "TruncatedTriangle.stl".
+        name (str, optional): defaults to "truncated_tri".
+        material_tag (str, optional): defaults to "truncated_tri_mat".
+    """
+
+    def __init__(
+        self,
+        l1,
```
Comment (Shimwell) [inline paramak/parametric_components/truncated_triangle.py]:
```suggestion
        length_1,
```

File: paramak/parametric_components/truncated_triangle.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        l2,
```
Comment (Shimwell) [inline paramak/parametric_components/truncated_triangle.py]:
```suggestion
        length_2,
```

File: paramak/parametric_components/truncated_triangle.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        l3,
```
Comment (Shimwell) [inline paramak/parametric_components/truncated_triangle.py]:
```suggestion
        length_3,
```

File: paramak/parametric_components/truncated_triangle.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+from paramak.utils import rotate
+
+
+class TruncatedTriangle(RotateStraightShape):
+    """Creates a rectangular poloidal field coil.
+
+    Args:
+        l1 (float): the (cm).
+        l2 (float): the horizontal (x axis) width of the coil (cm).
+        l3 (float): the center of the coil (x,z) values
```
Comment (Shimwell) [inline paramak/parametric_components/truncated_triangle.py]:
```suggestion
        length_1 (float): the (cm).
        length_2 (float): the horizontal (x axis) width of the coil (cm).
        length_3 (float): the center of the coil (x,z) values
```

File: tests/test_parametric_components/test_TruncatedTriangle.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+        self.test_shape.length_1=10
+        self.test_shape.length_2=42
+
+        assert self.test_shape.volume > test_shape_vol
+
+        self.test_shape.length_1=10
+        self.test_shape.length_2=20
+        self.test_shape.length_3=42
+
+        assert self.test_shape.volume > test_shape_vol
```
Comment (Shimwell) [inline tests/test_parametric_components/test_TruncatedTriangle.py:74]:
```suggestion
        assert self.test_shape.volume > test_shape_vol

```

**Review by Shimwell — COMMENTED** [review]
I think this is about ready

**Comment by RemDelaporteMathurin** [comment]
Cool very neat! Could this replace the rotated port cutter ?

File: paramak/parametric_components/truncated_triangle.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+
+class TruncatedTriangle(RotateStraightShape):
+    """Creates a rectangular poloidal field coil.
+
+    Args:
+        length_1 (float): the (cm).
+        length_2 (float): the horizontal (x axis) width of the coil (cm).
+        length_3 (float): the center of the coil (x,z) values
+        pivot_point (tuple of floats): the coordinates of the center of
```
Comment (RemDelaporteMathurin) [inline paramak/parametric_components/truncated_triangle.py:15]:
```suggestion
        pivot_point ((float, float)): the coordinates of the center of
```

**Comment by RemDelaporteMathurin** [comment]
Would it be clearer if this shape was called RotatedTrapeze instead ?

**Comment by Shimwell** [comment]
> Cool very neat! Could this replace the rotated port cutter ?

I think it could provide a nice alternative to the rotated port cutter. But it needs more args from the user. In the situation I want to use it in I am a bit lucky as I know the coords of the pivot point.

**Comment by Shimwell** [comment]
> Would it be clearer if this shape was called RotatedTrapeze instead ?

Always there with the great ideas. I've renamed :-)

**Pull request merged**